src/main.ts

```diff
@@ -0,0 +1,128 @@
+const core = require ('@actions/core');
+const path = require('path');
+const https = require('https');
+
+const jsonPath = path.join(__dirname, '..', 'resources', 'notification.json');
+const workflow = process.env.GITHUB_WORKFLOW;
+const repository = process.env.GITHUB_REPOSITORY;
+const branch = process.env.GITHUB_REF;
+const event_payload =  process.env.GITHUB_EVENT_PATH || jsonPath;
+let event_name = process.env.GITHUB_EVENT_NAME;
+
+let msteams_webhook_url;
+let job_status;
+let event_id;
+let details;
+let account;
+let message;
+let url;
+
+
+async function parse_inputs() {
+    try {
+
+        // do not proceed if jobs was cancelled
+        job_status = core.getInput('job_status');
+        if (job_status.toUpperCase() === "CANCELLED") {
+            console.log("Job was cancelled, no notification will be sent")
+            process.exit(0);
```

Comment:
if you chose to make this function pure, consider moving this conditional outside instead of terminating the process from withing the function.

something along the lines of:

```js
if (core.getInput('job_status').toUpperCase() !== 'CANCELLED') {
  const notifParams = parseInput(
    core.getInput('webhook_url'),
    core.getInput('event_id'),
    core.getInput('details')
  )
  notifyTeams(notifParams)
} else {
  console.log("Job was cancelled, no notification will be sent")
  process.exit(0);
}
```

Reply:
Good hint, the conditional has been moved out of the "parsing" function : 
```
if (core.getInput('job_status').toUpperCase() !== 'CANCELLED') {
    notifyTeams(parsedNotificationMessage(JSON.stringify(githubEventPayload)));
} else {
    core.warn('Job was cancelled, no notification will be sent');
    process.exit(0);
}
```